patches/tModLoader/Terraria.ModLoader/Logging.cs

```diff
@@ -188,21 +188,33 @@ public static class Logging
 			if (handlerActive.Value)
 				return;
 
+			bool oom = args.Exception is OutOfMemoryException;
+
+			//In case of OOM, unload the Main.tile array and do immediate garbage collection.
+			//If we don't do this, there will be a big chance that this method will fail to even quit the game, due to another OOM exception being thrown.
+			if (oom) {
+				Main.tile = null;
```

Comment:
Are there other kinds of unloading we can do here to free up more memory? (We should look into unloading all textures from items and such, since they are now loaded on demand)